### PR TITLE
Add fee and initiator to Pending Channels RPC

### PR DESCRIPTION
**Pull request by carlaKC**
This PR adds channel initiator and fee fields the the pending channels messages where they are missing. It also deprecates the `pending_close_channels` field, which was kept around to cover the case where somebody upgraded their node after cooperatively closing their channel, but before the close tx confirmed. The change was made in 2018, and pending close is only a temporary state, so it is acceptable to deprecate it. Adding channels to the deprecated field is replaced with a warning log, in case somebody with an old node upgrades and hits this edge case. 

Fixes: #4067

### CODE REVIEW

File: lnrpc/rpc.proto [hunk 1]
```diff
@@ -2028,6 +2028,24 @@ message PendingChannelsResponse {
 
         /// Hash of the remote pending version of the commitment tx.
         string remote_pending_txid = 3;
+
+        /*
+        The amount in satoshis calculated to be paid in fees for the local
+        commitment.
+        */
+        uint64 local_commit_fee = 4;
```
Comment (guggero) [inline lnrpc/rpc.proto]: nit: `_sat` suffix for all of these?

File: lnrpc/rpc.proto [hunk 2]
```diff
@@ -2096,8 +2096,7 @@ message PendingChannelsResponse {
     /// Channels pending opening
     repeated PendingOpenChannel pending_open_channels = 2;
 
-    /// Channels pending closing
-    repeated ClosedChannel pending_closing_channels = 3;
+    reserved 3;
```
Comment (guggero) [inline lnrpc/rpc.proto]: Not sure if we should first set it to `pending_closing_channels = 3 [deprecated = true];` first and just not fill it any more? Then remove it in the next version? Not sure what the `reserved` does for clients that use a previous RPC client?

File: rpcserver.go [hunk 3]
```diff
@@ -2697,6 +2698,29 @@ func (r *rpcServer) PendingChannels(ctx context.Context,
 			LocalBalance:  int64(pendingClose.SettledBalance),
 		}
 
+		// Lookup the channel in the historical channel bucket to obtain
+		// initiator information. If the historical channel bucket was
+		// not found, or the channel itself, this channel was closed
+		// in a verision before we started persisting historical
```
Comment (bhandras) [inline rpcserver.go]: nit: verision.

File: lnrpc/rpc.proto (hunk 1 above)
Comment (bhandras) [reply lnrpc/rpc.proto]: I think having these suffixes may make it harder to adopt msat (if we ever do that :)) as we'll need to deprecate a bunch of fields and introduce new ones. wdyt @guggero ?

File: lnrpc/rpc.proto (hunk 2 above)
Comment (bhandras) [reply lnrpc/rpc.proto]: Agree, I too think deprecating first is the better way to do this. (Can even leave it deprecated).

File: lnrpc/rpc.proto (hunk 1 above)
Comment (carlaKC) [reply lnrpc/rpc.proto]: Or go straight to msat with these? Feels conuterintuitive for values that come straight off a commitment, but if all msat fields is where we want to be (is it?), could just skip ahead to that.

Comment (guggero) [reply lnrpc/rpc.proto]: There will probably never be msats for this, as it's chain related. But a user wouldn't necessarily know this so the `_sat` suffix would be to make sure everyone understands what this value contains. I'm also fine with an explicit mention in the comment only. But I wouldn't convert it to msat, that wouldn't make sense really.

Comment (carlaKC) [reply lnrpc/rpc.proto]: Went with `_sat`, looks like we use it in a few other places :) Agree that msat would be odd here.

File: rpcserver.go [hunk 4]
```diff
@@ -2697,6 +2698,29 @@ func (r *rpcServer) PendingChannels(ctx context.Context,
...
+			&pendingClose.ChanPoint,
+		)
+		switch err {
+		// If the channel was closed in a version that did not record
+		// historical channels, ignore the error.
+		case channeldb.ErrNoHistoricalBucket:
+		case channeldb.ErrChannelNotFound:
+
+		case nil:
+			channel.Initiator = historical.IsInitiator
```
Comment (joostjager) [inline rpcserver.go]: Shouldn't the field distinguish between non-initiator and a legacy channel for which no history is present?

Comment (carlaKC) [reply rpcserver.go]: hm yes it should, at the time I thought we wouldn't have legacy channels in pending, but there is of course the old upgrade path option of a channel closing before upgrade, and list pending after. This should use the `Initiator` enum.

**Pull request merged**